src/context/context.ts

```diff
@@ -334,49 +340,47 @@ export class WorkflowContext<TInitialPayload = unknown> {
   }
 
   /**
-   * Makes the workflow run wait until a notify request is sent or until the
-   * timeout ends
+   * Pauses workflow execution until a specific event occurs or a timeout is reached.
    *
-   * ```ts
-   * const { eventData, timeout } = await context.waitForEvent(
-   *   "wait for event step",
-   *   "my-event-id",
-   *   100 // timeout after 100 seconds
-   * );
-   * ```
+   *```ts
+   * const result = await workflow.waitForEvent("payment-confirmed", {
+   *   timeout: "5m"
+   * });
+   *```
    *
-   * To notify a waiting workflow run, you can use the notify method:
+   * To notify a waiting workflow:
    *
    * ```ts
    * import { Client } from "@upstash/workflow";
    *
-   * const client = new Client({ token: });
+   * const client = new Client({ token: "<QSTASH_TOKEN>" });
    *
    * await client.notify({
-   *   eventId: "my-event-id",
-   *   eventData: "eventData"
+   *   eventId: "payment.confirmed",
+   *   data: {
+   *     amount: 99.99,
+   *     currency: "USD"
+   *   }
    * })
    * ```
    *
    * @param stepName
-   * @param eventId event id to wake up the waiting workflow run
-   * @param timeout timeout duration in seconds
-   * @returns wait response as `{ timeout: boolean, eventData: unknown }`.
-   *   timeout is true if the wait times out, if notified it is false. eventData
-   *   is the value passed to `client.notify`.
+   * @param eventId - Unique identifier for the event to wait for
+   * @param options - Configuration options
+   * @returns `{ timeout: boolean, eventData: unknown }`.
+   *   The `timeout` property specifies if the workflow has timed out. The `eventData`
+   *   is the data passed when notifying this workflow of an event.
    */
   public async waitForEvent(
     stepName: string,
     eventId: string,
-    timeout: number | Duration
+    options: WaitEventOptions = {}
   ): Promise<WaitStepResponse> {
-    const result = await this.addStep(
-      new LazyWaitForEventStep(
-        stepName,
-        eventId,
-        typeof timeout === "string" ? timeout : `${timeout}s`
-      )
-    );
+    const { timeout = 60 * 60 * 24 } = options;
```

Comment:
I think the timeout should be the user’s maximum quota for delay if it's not provided, as 24 hours might not be intuitive.

For example, if I send an email and the workflow resumes only when the user interacts with it, the developer would need to explicitly define the timeout value.

Setting it to the maximum quota feels more logical to me—at least making it the free users’ maximum, 7 days. WDYT?